### ISSUE TITLE
[BUG] Can't use external kernel modules with crc bundles

### ISSUE THREAD

**Issue by ybettan** [crc-org/crc#3737]
### General information

  * OS: Linux
  * Hypervisor: KVM
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop

## CRC version
```bash
ybettan kernel-module-management (main) $ crc version
CRC version: 2.20.0+f3a947
OpenShift version: 4.13.0
Podman version: 4.4.4
```
  
## CRC status
```bash
ybettan kernel-module-management (main) $ crc status --log-level debug
DEBU CRC version: 2.20.0+f3a947
DEBU OpenShift version: 4.13.0
DEBU Podman version: 4.4.4
DEBU Running 'crc status'
CRC VM:          Running
OpenShift:       Running (v4.13.0)
RAM Usage:       7.429GB of 9.369GB
Disk Usage:      19.74GB of 32.68GB (Inside the CRC VM)
Cache Usage:     37.92GB
Cache Directory: /home/ybettan/.crc/cache
```

## CRC config
```bash
ybettan kernel-module-management (main) $ crc config view
- consent-telemetry                     : no
```

## Host Operating System
```bash
ybettan kernel-module-management (main) $ cat /etc/os-release
NAME="Fedora Linux"
VERSION="36 (Workstation Edition)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora Linux 36 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
SUPPORT_END=2023-05-16
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

### Steps to reproduce

  1. crc setup
  2. crc start

### Expected
Kernel version of the node to be `5.14.0-284.13.1.el9_2.x86_64`

### Actual
Kernel version of the node is in fact `5.14.0-70.53.1.el9_0.x86_64`

### Context

My team is developing [KMM](https://github.com/rh-ecosystem-edge/kernel-module-management)  and seems like using CRC (instead of plane OCP) will not work with [DTK](https://github.com/openshift/driver-toolkit) which is built for every OCP z-stream with the specific kernel packages required for building kernel-modules for that kernel.

I have only tested it on CRC-4.13.0 but I believe the error will be the same on later version as well.
4.12.z seems to work OK though.

Adding here the link to the internal RedHat slack channel in case it adds more context: https://redhat-internal.slack.com/archives/C02G2TR52AX/p1687533542518989

**Comment by cfergeau** [comment]
https://github.com/crc-org/snc/issues/745 for the corresponding `snc` bug. I asked Yoni to file an issue in crc so that we are aware of the "external module" use case, which does not work well with kernel downgrades.

---

the wrong reference was stated.

**Comment by gbraad** [comment]
This is a very specific request and constitutes the use of 'custom' bundles.

You can derive and change the kernel and then use:
```
$ crc bundle generate
```
to create this 'custom' bundle

**Comment by qbarrand** [comment]
Hi @gbraad, the problem here is that the default CRC bundle for 4.13.0 does not ship the same kernel as OpenShift.
This breaks [driver-toolkit (DTK)](https://github.com/openshift/driver-toolkit) that we use to build kmods, because DTK is built for `5.14.0-284.13.1.el9_2.x86_64` while the CRC node runs `5.14.0-70.53.1.el9_0.x86_64`.
No custom bundle is involved.

**Comment by gbraad** [comment]
No, but you can create this for your own reuse.
Start CRC, login and make customizations, check and then export using generate.

The other approach is to use the snc path for this by using a custom payload, or install as usual, and befotre the bundling install your own kernel.

This is a very specific request that is not really a CRC issue, but for the snc side/bundles.

Note: a similar request is to include certain drivers (perhaps depend on a certain kernel) to run certain payloads, like AI, etc.

**Comment by qbarrand** [comment]
What is snc?
We do not want specific kernels here, rather we expect CRC to run the kernel that OpenShift is released with, so that DTK works as expected.

**Comment by gbraad** [comment]
`single node cluster` generation scripts: https://github.com/crc-org/snc

> specific kernels

Yes you do... as stated in the following sentence.

> we expect CRC to run the kernel that OpenShift is released with

which is very specific ;-). We run this kernel so that OpenShift can run on M1 and other platforms. Currently, this is not possible with the kernel OpenShift is provided with. For our developer use-case this is more important as this is the second-largest group of users.

As said, a 'customized' bundle is in that case necessary to prevent the kernel change. For the moment this is a necessity.

**Comment by gbraad** [comment]
The issue https://github.com/crc-org/snc/issues/745 exists which might address this. As that issue states that the downgrade/kernel modification should only happen for Aarch64. However, as stated before, you can work around this by using your own 'customized' bundles, which avoid changing the kernel/restores to the specific kernel you want.

**Comment by gbraad** [comment]
Hi @qbarrand @ybettan This has been taken care of in: https://github.com/crc-org/snc/pull/749

**Comment by cfergeau** [comment]
Do we keep this open to track the macos limitation? or do we close it?

**Comment by gbraad** [comment]
close this. if we need to track this, it is better to create a new issue that is more specific about the needs. it can reference this if needed.
